### PR TITLE
UCT/RC/VERBS: TM eager support (infra and receive ops)

### PR DESCRIPTION
**Pull request by alex-mikheev**

### CODE REVIEW

**Comment by mellanox-github** [comment]
Test PASSed.
See http://bgate.mellanox.com/jenkins/job/gh-ucx-pr/1434/ for details.

**Comment by mellanox-github** [comment]
Test FAILed.
See http://hpc-master.lab.mtl.com:8080/job/hpc-ucx-pr/3361/ for details (Mellanox internal link).

**Comment by yosefe** [comment]
bot:mlx:retest

File: src/uct/ib/rc/verbs/rc_verbs_iface.c [hunk 1]
```diff
@@ -134,72 +134,229 @@ void uct_rc_verbs_iface_progress(void *arg)
...
+                                          cqe->byte_len - tm_info_len,
+                                          UCT_CB_FLAG_DESC,
+                                          tm_info.tag.tag, imm_data);
+        if (status == UCS_OK) {
+            ucs_mpool_put_inline(ib_desc);
+        } else {
+            udesc = (char*)ib_desc + iface->tm.eager_desc.shift;
+            uct_recv_desc(udesc) = &iface->tm.eager_desc.super;
+        }
+        if (ucs_unlikely(!(++iface->tm.unexpected_cnt % IBV_DEVICE_MAX_UNEXP_COUNT))) {
```
Comment (yosefe) [inline src/uct/ib/rc/verbs/rc_verbs_iface.c]: seems wrong - dividing by (16k-1)

File: src/uct/ib/rc/verbs/rc_verbs_iface.c [hunk 2]
```diff
@@ -134,72 +134,229 @@ void uct_rc_verbs_iface_progress(void *arg)
...
+    uct_rc_verbs_ctx_priv_t *priv;
+    struct ibv_wc_tm_info tm_info;
+    int ret;
+
+    ret = ibv_start_poll(cq_ex, &cq_attr);
+    if (ret) {
+        if (ucs_likely(ret == ENOENT)) {
+            return UCS_ERR_NO_PROGRESS;
+        } else {
+            ucs_fatal("Failed to poll receive CQ %d", ret);
```
Comment (yosefe) [inline src/uct/ib/rc/verbs/rc_verbs_iface.c]: "ibv_start_poll(cq=%p) failed: %m"

File: src/uct/ib/rc/verbs/rc_verbs_iface.c [hunk 3]
```diff
@@ -235,8 +462,11 @@ static ucs_status_t uct_rc_verbs_iface_tag_init(uct_rc_verbs_iface_t *iface,
         srq_init_attr.pd                  = md->pd;
         srq_init_attr.cq                  = iface->super.super.recv_cq;
         srq_init_attr.tm_cap.max_num_tags = iface->tm.tag_available;
-        srq_init_attr.tm_cap.max_tm_ops   = ucs_min(2*iface->tm.tag_available,
-                                            IBV_DEVICE_TM_CAPS(&md->dev, max_tag_ops));
+
+        /* 2 ops for each tag (ADD + DEL) and extra 8 tags for SYNC ops.
+         * Do not rely on max_tag_ops device capability, so that XRQ creation
+         * would fail if such # of ops is not supported (which is not expected). */
+        srq_init_attr.tm_cap.max_tm_ops   = 2 * iface->tm.tag_available + 8;
```
Comment (yosefe) [inline src/uct/ib/rc/verbs/rc_verbs_iface.c]: where is the number '8' coming from? could it be calculated / #defined ? does it depend on IBV_DEVICE_MAX_UNEXP_COUNT?

**Comment by mellanox-github** [comment]
Test PASSed.
See http://hpc-master.lab.mtl.com:8080/job/hpc-ucx-pr/3372/ for details (Mellanox internal link).

**Comment by mellanox-github** [comment]
Test PASSed.
See http://bgate.mellanox.com/jenkins/job/gh-ucx-pr/1461/ for details.

**Comment by mellanox-github** [comment]
Test FAILed.
See http://hpc-master.lab.mtl.com:8080/job/hpc-ucx-pr/3384/ for details (Mellanox internal link).

**Comment by mellanox-github** [comment]
Test PASSed.
See http://hpc-master.lab.mtl.com:8080/job/hpc-ucx-pr/3387/ for details (Mellanox internal link).

**Pull request merged**